src/Compilers/CSharp/Test/Semantic/Semantics/TargetTypedDefaultTests.cs

```diff
@@ -711,14 +743,307 @@ class C
+                // (10,17): error CS8310: Operator '^' cannot be applied to operand 'default'
+                //         var e = default ^ default;
+                Diagnostic(ErrorCode.ERR_BadOpOnNullOrDefault, "default ^ default").WithArguments("^", "default").WithLocation(10, 17),
+                // (11,17): error CS8310: Operator '*' cannot be applied to operand 'default'
+                //         var f = default * default;
+                Diagnostic(ErrorCode.ERR_BadOpOnNullOrDefault, "default * default").WithArguments("*", "default").WithLocation(11, 17),
+                // (12,17): error CS8310: Operator '/' cannot be applied to operand 'default'
+                //         var g = default / default;
+                Diagnostic(ErrorCode.ERR_BadOpOnNullOrDefault, "default / default").WithArguments("/", "default").WithLocation(12, 17),
+                // (13,17): error CS8310: Operator '%' cannot be applied to operand 'default'
+                //         var h = default % default;
+                Diagnostic(ErrorCode.ERR_BadOpOnNullOrDefault, "default % default").WithArguments("%", "default").WithLocation(13, 17),
+                // (14,17): error CS8310: Operator '>>' cannot be applied to operand 'default'
+                //         var i = default >> default;
+                Diagnostic(ErrorCode.ERR_BadOpOnNullOrDefault, "default >> default").WithArguments(">>", "default").WithLocation(14, 17),
+                // (15,17): error CS8310: Operator '<<' cannot be applied to operand 'default'
+                //         var j = default << default;
+                Diagnostic(ErrorCode.ERR_BadOpOnNullOrDefault, "default << default").WithArguments("<<", "default").WithLocation(15, 17),
+                // (16,17): error CS8310: Operator '>' cannot be applied to operand 'default'
+                //         var k = default > default;
+                Diagnostic(ErrorCode.ERR_BadOpOnNullOrDefault, "default > default").WithArguments(">", "default").WithLocation(16, 17),
+                // (17,17): error CS8310: Operator '<' cannot be applied to operand 'default'
+                //         var l = default < default;
+                Diagnostic(ErrorCode.ERR_BadOpOnNullOrDefault, "default < default").WithArguments("<", "default").WithLocation(17, 17),
+                // (18,17): error CS8310: Operator '>=' cannot be applied to operand 'default'
+                //         var m = default >= default;
+                Diagnostic(ErrorCode.ERR_BadOpOnNullOrDefault, "default >= default").WithArguments(">=", "default").WithLocation(18, 17),
+                // (19,17): error CS8310: Operator '<=' cannot be applied to operand 'default'
+                //         var n = default <= default;
+                Diagnostic(ErrorCode.ERR_BadOpOnNullOrDefault, "default <= default").WithArguments("<=", "default").WithLocation(19, 17),
+                // (20,17): error CS0034: Operator '==' is ambiguous on operands of type 'default' and 'default'
```

Comment:
>// (20,17): error CS0034: Operator '==' is ambiguous on operands of type 'default' and 'default' [](start = 16, length = 96)

It looks like we need to follow-up on this error message separately or, perhaps, to change condition in BinaryOperatorOverloadResolution to always enter the ```if``` when both operands are ```default``` and handle that case specially in ReportBinaryOperatorError.